File: project/byeongjin-onboarding/src/main/java/com/example/byeongjin_onboarding/entity/Survey.java
package com.example.byeongjin_onboarding.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Survey {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(nullable = false)
    private LocalDateTime updatedAt;

    @OneToMany(mappedBy = "survey", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    @OrderBy("displayOrder ASC")
    private List<FormItem> formItems = new ArrayList<>();

    public void addFormItem(FormItem formItem) {
        formItems.add(formItem);
        formItem.setSurvey(this);
    }

    public void updateFormItems(List<FormItem> newFormItems) {
        this.formItems.clear();
        if (newFormItems != null) {
            for (FormItem newItem : newFormItems) {
                this.addFormItem(newItem);
            }
        }
    }
}

File: project/kyungrok-onboarding/src/main/kotlin/kr/innercircle/onboarding/survey/domain/Survey.kt
package kr.innercircle.onboarding.survey.domain

import jakarta.persistence.*
import kr.innercircle.onboarding.survey.dto.request.CreateSurveyRequest
import kr.innercircle.onboarding.survey.dto.request.UpdateSurveyRequest

/**
 * packageName : kr.innercircle.onboarding.survey.domain
 * fileName    : Survey
 * author      : ckr
 * date        : 25. 6. 11.
 * description :
 */

@Table(name = "survey")
@Entity
class Survey (
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(nullable = false, length = 200)
    var name: String,

    @Column(columnDefinition = "TEXT")
    var description: String? = null
): BaseTimeEntity() {
    fun update(updateSurveyRequest: UpdateSurveyRequest) {
        this.name = updateSurveyRequest.name
        this.description = updateSurveyRequest.description
    }

    companion object {
        fun from(createSurveyRequest: CreateSurveyRequest): Survey {
            return Survey(
                name = createSurveyRequest.name,
                description = createSurveyRequest.description
            )
        }
    }
}